### ISSUE TITLE
Alarm fired!

### ISSUE THREAD

**Issue by bulosh** [Minims/SomfyProtect2MQTT#6]
My Somfy Protect alarm doesn't tell me when is ringing/is fired..
is it a future feature or mine is not working?

**Comment by Minims** [comment]
Hello,

I have not test this part for now. So it a feature/fix to make.
I think I have not the right payload for HA.
Do you have the alarm status returned that has been push to MQTT when it was fired ? 
By default it's supposed to be in this topic : somfyProtect2mqtt/<zone_id>/state

**Comment by Minims** [comment]
@bulosh Can you help to debug this issue ? :-)

**Comment by Shkrum31** [comment]
@Minims Thanks for this addon.

It seems a lot more reliable than the integration via ifttt!

About the highlighted issue, I do confirm that the `switch.turn_on` and `switch.turn_off` services applied to the `switch.maison_siren` entity work perfectly to trigger or stop the siren.
However, there is no feedback about the siren state from Somfy to HA.

As a result:
- After triggering the siren from HA (`switch.turn_on` applied to the `switch.maison_siren` entity), the switch state changes to `on`. However, if one has stopped the siren directly from the Somfy app, or after the 3 minutes of ringing are over and the siren has automatically stopped, the state of the switch stays `on`
- After triggering the siren from the Somfy app, the state of the `switch.maison_siren` entity does not change.

I do not know if it would be possible to « sync » the siren actual state with the switch entity within HA.

If you need any help, I can to help you test and debug your code!

**Comment by Minims** [comment]
> However, there is no feedback about the siren state from Somfy to HA.

I need to dig in the somfy api, for now i can't see where the status is updated.
 
> As a result:
> 
> * After triggering the siren from HA (`switch.turn_on` applied to the `switch.maison_siren` entity), the switch state changes to `on`. However, if one has stopped the siren directly from the Somfy app, or after the 3 minutes of ringing are over and the siren has automatically stopped, the state of the switch stays `on`

Ok, i might set the switch as temporary on if I can't check the status in the API.

> * After triggering the siren from the Somfy app, the state of the `switch.maison_siren` entity does not change.
> 
> I do not know if it would be possible to « sync » the siren actual state with the switch entity within HA.
> 
> If you need any help, I can to help you test and debug your code!

Will look how to do when I have some time to work on it. 
I saw an other project and they status are taken from IFTTT :-/

**Comment by Shkrum31** [comment]
Thanks for your answer.

> I saw an other project and they status are taken from IFTTT :-/

This is indeed what I do: I mitigate the problem via IFTTT. Contrary to the configuration described in the [alarm_control_panel.ifttt](https://www.home-assistant.io/integrations/alarm_control_panel.ifttt/) integration, only one recipe in IFTTT is required any more with your add-on: a recipe that starts a 3 minutes timer entity within Home Assistant when the alarm rings (either automatically when triggered if previously armed, or manually via the Somfy app).

> Ok, i might set the switch as temporary on if I can't check the status in the API.

Indeed, it would be really nice that after 3 minutes at the `on` state, the switch automatically changes to `off`.

In this case, my timer entity would not be necessary anymore.
However, in such scenario, in order to avoid any unwanted loops : 
- If the `switch.maison_siren` entity changes via the `switch.turn_off` or `switch.turn_on` services directly from Home Assistant, some actions are required for the Somfy System to react accordingly (this is what your add-on currently does perfectly).
- If the siren triggers from the Somfy System, and an IFTTT applet triggers to make the `switch.maison_siren` change from `off` to `on` to reflect the assumed Somfy siren state, no further action is required.
- If after 3 minutes at the `on` state, the `switch.maison_siren` automatically changes to `off`, no further action is neither required (as the siren of the Somfy System would already be off)

**Comment by Minims** [comment]
@Shkrum31 I'm looking for a way to integrate the "temporary" switch with a 3min lifetime. but I can't find a solution with HADiscovery : https://www.home-assistant.io/docs/mqtt/discovery/
Any Idea what kind of entity I can configure for that ?

**Comment by rpartapsing** [comment]
Hi, any updates on this?

**Comment by Minims** [comment]
I'm working on big update, I need to rework a big part a the project.
It will add : 
 - immediate device precence notification
 - immediate alram status notification, including alarm is ringing
 - Test mode.
maybe some other thing, it depends what I will discover in the next steps.

**Comment by Minims** [comment]
Update pushed on dev branch. 
This is an alpha release, not fully tested. 
I have add : 

- immediate device precence notification
- immediate alram status notification, including alarm is ringing

**Comment by bulosh** [comment]
> Update pushed on dev branch. This is an alpha release, not fully tested. I have add :
> 
> * immediate device precence notification
> * immediate alram status notification, including alarm is ringing

Can’t find in HA these functions…where can I find TheM as entities?

**Comment by Minims** [comment]
About "immediate alram status notification, including alarm is ringing" , it is just the alarm device that is update in few seconds.
About "immediate device precence notification", there is a new device_tracker  entity in each Badge called : "Badge de XXXXX presence"
Starting status is unkown as I need to receive a message from smoky on web socket to update it.

**Comment by Shkrum31** [comment]
Thanks @Minims for maintening this add-on.
So that I can help you test and debug, can you make your modifications available as a contributor through the https://github.com/schumijo/homeassistant-addons/ repo, or instruct me another way ?
On my system, the add-on is installed via Supervisor within assio in a VM.
Thanks

**Comment by Minims** [comment]
Hello, you can use this add-on. Install the dev version. If you still have it installed. Click on rebuild/recontruire in the add on menu to get tje latest dev version

**Comment by Shkrum31** [comment]
Sorry, but are you sure the following is your fork ?

![image](https://user-images.githubusercontent.com/63510161/158028643-aef15223-685b-4183-8f30-ccb53253c124.png)

**Comment by Minims** [comment]
Yes schumijo make this when I release the first version.
There is my fork too that a I use for my dev : https://github.com/Minims/homeassistant-addons
but I always sync it with schumijo when needed.
Here it's not needed for the dev version as there is no change in the config file.

**Comment by Shkrum31** [comment]
Thanks ! I wills provide some feedback

**Comment by Shkrum31** [comment]
What I tested so far about the triggering of the alarm :

- When panic button is used from the Somfy app in order to trigger the alarm manually, an event occurs within HA : the alarm state changes to « triggered » then changes back to its previous state (« disarmed » for exemple) a few seconds after. If this is what you intended to do, good job 👍 

- However, when triggered by detection (intellitag or motion), no event occurs within HA. IFTTT is still required.

**Comment by Minims** [comment]
> 



> * When panic button is used from the Somfy app in order to trigger the alarm manually, an event occurs within HA : the alarm state changes to « triggered » then changes back to its previous state (« disarmed » for exemple) a few seconds after. If this is what you intended to do, good job 👍
> *

Yes :-)



> * However, when triggered by detection (intellitag or motion), no event occurs within HA. IFTTT is still required.

Ok, So I think a i must received a message on websocket for this.
If you have debug mode enable, you might have an unknown message on websocket  matching the triggered alarm.
If you retrieve the log message in HA addons, on in the docker container, it would be great.

**Comment by Shkrum31** [comment]
After looking at the log messages, please find below some findings:



### Some error messages
The two following lines appeared together every times :
```
[ERROR] [websocket:65] error from callback <bound method SomfyProtectWebsocket.on_message of <somfy_protect.websocket.SomfyProtectWebsocket object at 0x7f24e182b9d0>>: name 'site' is not defined
[ERROR] [somfy_protect.websocket:91] Error in the websocket connection: name 'site' is not defined
```

### Alarm Status changed by Somfy App :
```
[DEBUG] [websocket:78] ++Rcv raw: b'\x81~\x00\xdd{"profiles":["owner","admin","guest","kid"],"site_id":"XXX","type":"config","key":"security.level.change","security_level":"armed","message_id":"XXX"}'
[DEBUG] [websocket:78] ++Rcv decoded: fin=1 opcode=1 data=b'{"profiles":["owner","admin","guest","kid"],"site_id":"XXX","type":"config","key":"security.level.change","security_level":"armed","message_id":"XXX"}'
[DEBUG] [root:63] Message: {"profiles":["owner","admin","guest","kid"],"site_id":"XXX","type":"config","key":"security.level.change","security_level":"armed","message_id":"XXX"}
```
Which looks ok !


### Detection via motion while alarm "armed" :
```
[DEBUG] [websocket:78] ++Rcv raw: b'\x81~\x01\xcf{"profiles":["owner","admin","custom","family","neighbor"],"site_id":"XXX","type":"alarm","key":"alarm.trespass","device_id":"XXX","device_type":"pir","start_at":"2022-03-14T17:17:12.000000Z","start_siren_at":"2022-03-14T17:17:42.000000Z","end_at":"2022-03-14T17:20:42.000000Z","end_siren_at":"2022-03-14T17:20:42.000000Z","manual_alarm":false,"message_id":"XXX"}'
[DEBUG] [websocket:78] ++Rcv decoded: fin=1 opcode=1 data=b'{"profiles":["owner","admin","custom","family","neighbor"],"site_id":"XXX","type":"alarm","key":"alarm.trespass","device_id":"XXX","device_type":"pir","start_at":"2022-03-14T17:17:12.000000Z","start_siren_at":"2022-03-14T17:17:42.000000Z","end_at":"2022-03-14T17:20:42.000000Z","end_siren_at":"2022-03-14T17:20:42.000000Z","manual_alarm":false,"message_id":"XXX"}'
[DEBUG] [root:63] Message: {"profiles":["owner","admin","custom","family","neighbor"],"site_id":"XXX","type":"alarm","key":"alarm.trespass","device_id":"XXX","device_type":"pir","start_at":"2022-03-14T17:17:12.000000Z","start_siren_at":"2022-03-14T17:17:42.000000Z","end_at":"2022-03-14T17:20:42.000000Z","end_siren_at":"2022-03-14T17:20:42.000000Z","manual_alarm":false,"message_id":"XXX"}
[DEBUG] [somfy_protect.websocket:87] Unknown message: {"profiles":["owner","admin","custom","family","neighbor"],"site_id":"XXX","type":"alarm","key":"alarm.trespass","device_id":"XXX","device_type":"pir","start_at":"2022-03-14T17:17:12.000000Z","start_siren_at":"2022-03-14T17:17:42.000000Z","end_at":"2022-03-14T17:20:42.000000Z","end_siren_at":"2022-03-14T17:20:42.000000Z","manual_alarm":false,"message_id":"XXX"}
```
I think the above is the message you are looking for 👍 
Please note that the `alarm.trespass` event is associated with a `device_type` and timing values :
- `start_at` (time of detection, i.e. 17:17:12)
- `start_siren_at` (time of detection + 30 seconds which is the delay I set before ringing, i.e. 17:17:42 )
- `end_at` = `end_siren_at` (after 3 minutes with siren ringing, i.e. 17:20:42) 



### Two series of logs when siren stopped manually by user via Somfy App :
```
[DEBUG] [websocket:78] ++Rcv raw: b'\x81~\x01v{"profiles":["owner","admin","custom","family","neighbor"],"site_id":"XXX","type":"alarm","key":"alarm.end","device_id":null,"device_type":null,"end_at":"2022-03-14T17:19:22.000000Z","end_siren_at":"2022-03-14T17:19:22.000000Z","stopped_by_user_id":"XXX","message_id":"XXX"}'
[DEBUG] [websocket:78] ++Rcv decoded: fin=1 opcode=1 data=b'{"profiles":["owner","admin","custom","family","neighbor"],"site_id":"XXX","type":"alarm","key":"alarm.end","device_id":null,"device_type":null,"end_at":"2022-03-14T17:19:22.000000Z","end_siren_at":"2022-03-14T17:19:22.000000Z","stopped_by_user_id":"XXX","message_id":"XXX"}'
[DEBUG] [root:63] Message: {"profiles":["owner","admin","custom","family","neighbor"],"site_id":"XXX","type":"alarm","key":"alarm.end","device_id":null,"device_type":null,"end_at":"2022-03-14T17:19:22.000000Z","end_siren_at":"2022-03-14T17:19:22.000000Z","stopped_by_user_id":"XXX","message_id":"XXX"}
```
```
[DEBUG] [websocket:78] ++Rcv raw: b'\x81~\x01]{"profiles":["owner","admin","custom","family","neighbor"],"site_id":"XXX","type":"alarm","key":"alarm.end","device_id":null,"device_type":null,"end_at":"2022-03-14T17:19:22.000000Z","end_siren_at":null,"stopped_by_user_id":"XXX","message_id":"XXX"}'
[DEBUG] [websocket:78] ++Rcv decoded: fin=1 opcode=1 data=b'{"profiles":["owner","admin","custom","family","neighbor"],"site_id":"XXX","type":"alarm","key":"alarm.end","device_id":null,"device_type":null,"end_at":"2022-03-14T17:19:22.000000Z","end_siren_at":null,"stopped_by_user_id":"XXX","message_id":"XXX"}'
[DEBUG] [root:63] Message: {"profiles":["owner","admin","custom","family","neighbor"],"site_id":"XXX","type":"alarm","key":"alarm.end","device_id":null,"device_type":null,"end_at":"2022-03-14T17:19:22.000000Z","end_siren_at":null,"stopped_by_user_id":"XXX","message_id":"XXX"}
```



### Use of Panic button with Somfy App while alarm "disarmed" :
```
[DEBUG] [websocket:78] ++Rcv raw: b'\x81~\x01\xad{"profiles":["owner","admin","custom","family","neighbor"],"site_id":"XXX","type":"alarm","key":"alarm.panic","device_id":null,"device_type":null,"start_at":"2022-03-14T17:21:07.000000Z","start_siren_at":"2022-03-14T17:21:07.000000Z","end_at":"2022-03-14T17:24:07.000000Z","end_siren_at":"2022-03-14T17:24:07.000000Z","manual_alarm":false,"message_id":"XXX"}'
[DEBUG] [websocket:78] ++Rcv decoded: fin=1 opcode=1 data=b'{"profiles":["owner","admin","custom","family","neighbor"],"site_id":"XXX","type":"alarm","key":"alarm.panic","device_id":null,"device_type":null,"start_at":"2022-03-14T17:21:07.000000Z","start_siren_at":"2022-03-14T17:21:07.000000Z","end_at":"2022-03-14T17:24:07.000000Z","end_siren_at":"2022-03-14T17:24:07.000000Z","manual_alarm":false,"message_id":"XXX"}'
[DEBUG] [root:63] Message: {"profiles":["owner","admin","custom","family","neighbor"],"site_id":"XXX","type":"alarm","key":"alarm.panic","device_id":null,"device_type":null,"start_at":"2022-03-14T17:21:07.000000Z","start_siren_at":"2022-03-14T17:21:07.000000Z","end_at":"2022-03-14T17:24:07.000000Z","end_siren_at":"2022-03-14T17:24:07.000000Z","manual_alarm":false,"message_id":"XXX"}
```
Here, there is no 30s delay before siren the rings. 
- `start_at` = `start_siren_at` (time of detection, i.e. 17:21:07)
- `end_at` = `end_siren_at` (after 3 minutes with siren ringing, i.e. 17:24:07) 

I hope that will help improve your code !

**Comment by Minims** [comment]
Many Thanks for your debugging @Shkrum31. I don't really have time to check all that case.



> ### Some error messages
> The two following lines appeared together every times :
> 
> ```
> [ERROR] [websocket:65] error from callback <bound method SomfyProtectWebsocket.on_message of <somfy_protect.websocket.SomfyProtectWebsocket object at 0x7f24e182b9d0>>: name 'site' is not defined
> [ERROR] [somfy_protect.websocket:91] Error in the websocket connection: name 'site' is not defined
> ```

Don't know why for now. it need more investigation in the code.



> ### Detection via motion while alarm "armed" :
> ```
> [DEBUG] [websocket:78] ++Rcv raw: b'\x81~\x01\xcf{"profiles":["owner","admin","custom","family","neighbor"],"site_id":"XXX","type":"alarm","key":"alarm.trespass","device_id":"XXX","device_type":"pir","start_at":"2022-03-14T17:17:12.000000Z","start_siren_at":"2022-03-14T17:17:42.000000Z","end_at":"2022-03-14T17:20:42.000000Z","end_siren_at":"2022-03-14T17:20:42.000000Z","manual_alarm":false,"message_id":"XXX"}'
> [DEBUG] [websocket:78] ++Rcv decoded: fin=1 opcode=1 data=b'{"profiles":["owner","admin","custom","family","neighbor"],"site_id":"XXX","type":"alarm","key":"alarm.trespass","device_id":"XXX","device_type":"pir","start_at":"2022-03-14T17:17:12.000000Z","start_siren_at":"2022-03-14T17:17:42.000000Z","end_at":"2022-03-14T17:20:42.000000Z","end_siren_at":"2022-03-14T17:20:42.000000Z","manual_alarm":false,"message_id":"XXX"}'
> [DEBUG] [root:63] Message: {"profiles":["owner","admin","custom","family","neighbor"],"site_id":"XXX","type":"alarm","key":"alarm.trespass","device_id":"XXX","device_type":"pir","start_at":"2022-03-14T17:17:12.000000Z","start_siren_at":"2022-03-14T17:17:42.000000Z","end_at":"2022-03-14T17:20:42.000000Z","end_siren_at":"2022-03-14T17:20:42.000000Z","manual_alarm":false,"message_id":"XXX"}
> [DEBUG] [somfy_protect.websocket:87] Unknown message: {"profiles":["owner","admin","custom","family","neighbor"],"site_id":"XXX","type":"alarm","key":"alarm.trespass","device_id":"XXX","device_type":"pir","start_at":"2022-03-14T17:17:12.000000Z","start_siren_at":"2022-03-14T17:17:42.000000Z","end_at":"2022-03-14T17:20:42.000000Z","end_siren_at":"2022-03-14T17:20:42.000000Z","manual_alarm":false,"message_id":"XXX"}
> ```
> 
> I think the above is the message you are looking for 👍 Please note that the `alarm.trespass` event is associated with a `device_type` and timing values :
> 
> * `start_at` (time of detection, i.e. 17:17:12)
> * `start_siren_at` (time of detection + 30 seconds which is the delay I set before ringing, i.e. 17:17:42 )
> * `end_at` = `end_siren_at` (after 3 minutes with siren ringing, i.e. 17:20:42)

So I have add the basic for `alarm.trespass`. It must trigger the alarm in HA now with latest dev version.
I think I can create a binary_sensor later for Intellitag & Pir as a motion sensor. But I will only works if the alarm is active.
Don't know how to manage time received for now for all the message.

All Changes : https://github.com/Minims/SomfyProtect2MQTT/pull/29

**Comment by Minims** [comment]
Fixed in release 0.1.7
